HighwayPatrol.h

```diff
@@ -1,11 +1,22 @@
 #pragma once 
 
 #include "Vehicle.h"
+#include <string>
+
+struct Vehicle;
+struct Highway;
 
 struct HighwayPatrol : Vehicle 
 {
     HighwayPatrol();
+    ~HighwayPatrol() override;
+    HighwayPatrol(const HighwayPatrol&);
+    HighwayPatrol& operator=(const HighwayPatrol&);
+
     void scanHighway(Highway* h);
 
     void pullOver( Vehicle* v, bool willArrest, Highway* h );
-};
\ No newline at end of file
+
+    std::string vehicleClass;
```

Comment:
This variable `vehicleClass` is only used inside that `if()` block in that `pullOver` function.
you do not need to extend its scope/lifetime to match that of the class.
make it a local variable in that function
- [ ] revise